From b5d70ff242df617ab8078b69086e59428fd4a53a Mon Sep 17 00:00:00 2001
From: Dan Callaghan <dcallagh@redhat.com>
Date: Tue, 19 Aug 2014 08:19:11 +1000
Subject: [PATCH 2/2] relax version requirements

---
 requirements_test.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/requirements_test.txt b/requirements_test.txt
index 3e2bb35..6e4ab61 100644
--- a/requirements_test.txt
+++ b/requirements_test.txt
@@ -1,2 +1,2 @@
-pytest==2.6.3
-pytest-cov==1.7.0
+pytest
+pytest-cov
-- 
1.9.3

