From ad9506de29e6547d61b3b68e4f39bb36a573615f Mon Sep 17 00:00:00 2001
From: Dan Callaghan <dcallagh@redhat.com>
Date: Wed, 2 Jan 2013 11:06:56 +1000
Subject: [PATCH] ArgumentParser no longer accepts version in Python 3.3

---
 gunicorn/config.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/gunicorn/config.py b/gunicorn/config.py
index 5e2d80a..f1a0870 100644
--- a/gunicorn/config.py
+++ b/gunicorn/config.py
@@ -67,10 +67,13 @@ class Config(object):
     def parser(self):
         kwargs = {
             "usage": self.usage,
-            "version": "%(prog)s (version " +  __version__ + ")\n",
             "prog": self.prog
         }
         parser = argparse.ArgumentParser(**kwargs)
+        parser.add_argument("-v", "--version",
+                action="version", default=argparse.SUPPRESS,
+                version="%(prog)s (version " +  __version__ + ")\n",
+                help="show program's version number and exit")
         parser.add_argument("args", nargs="*", help=argparse.SUPPRESS)
 
         keys = list(self.settings)
-- 
1.7.11.7

